crates/vm/vm.rs

```diff
@@ -75,34 +75,172 @@ impl EvmState {
     }
 }
 
-/// Executes all transactions in a block and returns their receipts.
-pub fn execute_block(block: &Block, state: &mut EvmState) -> Result<Vec<Receipt>, EvmError> {
-    let block_header = &block.header;
-    let spec_id = spec_id(&state.chain_config()?, block_header.timestamp);
-    //eip 4788: execute beacon_root_contract_call before block transactions
-    if block_header.parent_beacon_block_root.is_some() && spec_id == SpecId::CANCUN {
-        beacon_root_contract_call(state, block_header, spec_id)?;
-    }
-    let mut receipts = Vec::new();
-    let mut cumulative_gas_used = 0;
-
-    for transaction in block.body.transactions.iter() {
-        let result = execute_tx(transaction, block_header, state, spec_id)?;
-        cumulative_gas_used += result.gas_used();
-        let receipt = Receipt::new(
-            transaction.tx_type(),
-            result.is_success(),
-            cumulative_gas_used,
-            result.logs(),
-        );
-        receipts.push(receipt);
-    }
+cfg_if::cfg_if! {
+    if #[cfg(feature = "levm")] {
+        use ethereum_rust_levm::{
+            db::{Cache, Database as LevmDatabase},
+            errors::{TransactionReport, TxResult, VMError},
+            vm::VM,
+            Environment,
+        };
+        use std::{collections::HashMap, sync::Arc};
+        use ethereum_rust_core::types::{code_hash, TxType};
+
+        /// Executes all transactions in a block and returns their receipts.
+        pub fn execute_block(
+            block: &Block,
+            state: &mut EvmState,
+        ) -> Result<(Vec<Receipt>, Vec<AccountUpdate>), EvmError> {
+            let block_header = &block.header;
+            let spec_id = spec_id(&state.chain_config()?, block_header.timestamp);
+            //eip 4788: execute beacon_root_contract_call before block transactions
+            if block_header.parent_beacon_block_root.is_some() && spec_id == SpecId::CANCUN {
+                beacon_root_contract_call(state, block_header, spec_id)?;
+            }
+            let mut receipts = Vec::new();
+            let mut cumulative_gas_used = 0;
+
+            let store_wrapper = Arc::new(StoreWrapper {
+                store: state.database().unwrap().clone(),
```

Comment:
This function returns result, handle this unwrap properly.

Reply:
Same as https://github.com/lambdaclass/lambda_ethereum_rust/pull/1029/files#r1829787920